### PR TITLE
Move CompositeCompliance tests to their own TestCase

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #74646
* #74645
* __->__ #74644

This is in preparation for me adding additional tests for:
1. composite compliance of autograd formulas
2. composite compliance of forward-mode AD formulas

This PR also changes these tests to run on both CPU and CUDA. Previously
they were just run on CPU, but it turns out there's a lot of branching
on the device in composite operations in PyTorch today :/

Test Plan:
- wait for tests

Differential Revision: [D35186861](https://our.internmc.facebook.com/intern/diff/D35186861)

### CODE REVIEW

**Comment by zou3519** [comment]
Context for reviewers: I'm beefing up composite compliance testing that was introduced as https://github.com/pytorch/pytorch/pull/65819 and retagging all of you as reviewers.

File: test/test_ops.py [hunk 1]
```diff
@@ -678,24 +678,6 @@ def _test_inplace_preserve_storage(samples, variants):
...
 
-    # Checks if the operator (if it is composite) is written to support most
-    # backends and Tensor subclasses. See "CompositeImplicitAutograd Compliance"
-    # in aten/src/ATen/native/README.md for more details
-    #
-    # NB: onlyCPU because CompositeImplicitAutograd ops go through the same
-    # codepath on all devices. Ideally we'd use a meta device here but coverage
-    # for that is not good yet.
-    @unittest.skipIf(IS_FBCODE or IS_SANDCASTLE, '__torch_dispatch__ does not work in fbcode')
-    @onlyCPU
```
[the remark this answers is not in the document]
Comment (zou3519) [reply test/test_ops.py:689]: Numbers on my machine:
33s: `pytest test/test_ops.py -v -k "composite" `
12s: `pytest test/test_ops.py -v -k "composite and cpu"` (aka, the state before)

**Review by albanD — APPROVED** [review]
SGTM then

**Comment by zou3519** [comment]
@zou3519 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35186861).

**Pull request closed without merging**